### DISCUSSION TITLE
Layout customization

### DISCUSSION THREAD

**Discussion by seanlinsley** [activeadmin/activeadmin#7885] [Ideas]
After speaking with @vasilakisfil in [IRC](https://botbot.me/freenode/activeadmin/msg/12577261/), it became obvious that we need a built-in way of adding custom HTML to AA's main layout (particularly because [some gems](https://github.com/assembler/attachinary) use this to inject dynamic JS objects). I suggested this monkey patch for now:

``` ruby
# The Active Admin equivalent of putting this in your application layout:
# <head>
#   <%= cloudinary_js_config %>
# </head>

module ActiveAdmin
  module Views
    module Pages
      class Base < Arbre::HTML::Document

        alias_method :original_build_head, :build_active_admin_head

        def build_active_admin_head
          original_build_head

          within @head do
            text_node include_gon
          end
        end

      end
    end
  end
end

```

**Comment by jarinudom** [comment]
This would be handy, but another way to do it would be to inject the html in Rack.

**Comment by timoschilling** [comment]
What's about supporting the Deface Gem?
https://rubygems.org/gems/deface

**Comment by jarinudom** [comment]
I kind of like the idea of having something like an automatically included `app/views/_includes.html.(haml|erb)` and maybe putting `yield :head` right before `</head>` and `yield :footer` right below the `</body>` tag. That way we can just `content_for :head` etc. to inject javascript or whatever else.

**Comment by DimaSamodurov** [comment]
What about Idea of https://github.com/bys-control/activeadmin_custom_layout ? Would AA benefit from having page components API public and make it possible to define custom layout?  
E.g. it would allow rendering AA features like index_as_table with filtering within default application layout.
